client/src/components/LoginSession/tests/LoginSessionContainer-test.js

```diff
@@ -28,40 +39,153 @@ window.ss.config = {
+        let wrapper = shallow(<LoginSessionContainer {...props} />);
+        let loginSession = wrapper.find(LoginSession).first();
+
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: false,
+            failed: false,
+            submitting: false
+        });
+
+        const logoutRequest = loginSession.props().logout();
+
+        wrapper = wrapper.update();
+        loginSession = wrapper.find(LoginSession).first();
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: false,
+            failed: false,
+            submitting: true
+        });
+
+        httpResolve({
+            error: false,
+            success: global.success,
+            message: 'amazing success'
+        });
+        await logoutRequest;
+
+        wrapper = wrapper.update();
+        loginSession = wrapper.find(LoginSession).first();
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: true,
+            failed: false,
+            submitting: false
+        });
+
+        expect(props.displayToastSuccess).toBeCalledWith('amazing success');
+        expect(props.displayToastFailure).not.toBeCalled();
     });
 
     it('should call displayToastFailure on failure', async () => {
         setEndpointFetcherSuccess(false);
-        wrapper = shallow(<LoginSessionContainer {...props} />);
-        await wrapper.instance().logout();
-        expect(props.displayToastSuccess).toBeCalledTimes(0);
-        expect(props.displayToastFailure).toBeCalledTimes(1);
+        let wrapper = shallow(<LoginSessionContainer {...props} />);
+        let loginSession = wrapper.find(LoginSession).first();
+
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: false,
+            failed: false,
+            submitting: false
+        });
+
+        const logoutRequest = loginSession.props().logout();
+
+        wrapper = wrapper.update();
+        loginSession = wrapper.find(LoginSession).first();
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: false,
+            failed: false,
+            submitting: true
+        });
+
+        httpResolve({
+            error: true,
+            success: global.success,
+            message: 'horrible failure'
+        });
+        await logoutRequest;
+
+        wrapper = wrapper.update();
+        loginSession = wrapper.find(LoginSession).first();
+        expect(loginSession.props()).toMatchObject({
+            ...sessionData,
+            complete: true,
+            failed: true,
+            submitting: false
+        });
+
+        expect(props.displayToastFailure).toBeCalledWith('horrible failure');
+        expect(props.displayToastSuccess).not.toBeCalled();
+    });
+
+    it('Handles HTTP Request failure', async () => {
```

Comment:
Our initial test was only testing a regular failure. We were not testing a scenario where we had an actual HTTP failure.